packages/core/admin/server/controllers/license.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const fs = require('fs');
+// eslint-disable-next-line node/no-extraneous-require
+const ee = require('@strapi/strapi/lib/utils/ee');
+
+module.exports = {
+  async findLicense(ctx) {
+    let licenseInfo = {};
+
+    if (strapi.EE) {
+      licenseInfo = ee.licenseInfo;
+    } else {
+      licenseInfo = null;
+    }
+
+    // Make the license field disabled if not in development env
+    if (process.env.NODE_ENV !== 'development') {
+      return (ctx.body = {
+        data: {
+          messageId: 'Settings.license.wrong-env',
+          disabled: true,
+          licenseInfo,
+        },
+      });
+    }
+
+    // License file path
+    const licenseFilePath = './license.txt';
+
+    // Default messageId & activationType
+    let messageId = 'Settings.license.no-license';
+
+    if (process.env.STRAPI_LICENSE) {
+      messageId = 'Settings.license.strapi-license.found';
+
+      if (process.env.STRAPI_LICENSE == '') {
+        messageId = 'Settings.license.strapi-license.empty';
+      }
+    }
+
+    try {
+      // If the file already exists
+      if (fs.existsSync(licenseFilePath)) {
+        messageId = 'Settings.license.license-file.found';
+
+        const data = fs.readFileSync(licenseFilePath, 'utf8');
+
+        if (data == '') {
+          messageId = 'Settings.license.license-file.empty';
+        }
+      }
+    } catch (err) {
+      console.error(err);
+    }
+
+    ctx.body = {
+      data: {
+        messageId,
+        disabled: false,
+        licenseInfo,
+      },
+    };
+  },
+
+  async generateLicenseFile(ctx) {
+    strapi.reload.isWatching = false;
+
+    let result = { messageId: 'Settings.license.created', status: 'ok' };
+
+    const {
+      body: { license },
+    } = ctx.request;
+
+    try {
+      fs.writeFileSync('license.txt', license, error => {
+        if (error) throw error;
+      });
```

Comment:
I believe we need to think about the creation a file a bit more: it is not uncommon in environments like e.g. render to only sync/ re-mount one folder between deployments (e.g. public), which would cause the license.txt to be lost after re-deploying.

I believe the primary way to configure applications in cloud contexts today is through environment variables and we should promote it. What has been done in other contexts of Strapi was to print a message in the users shell, explaining how to set the env-variable properly in case the write fails (e.g. for read-only filesystems).

In this case, what I'd love to see is, that we in *any case* display the env variable somewhere in the admin UI for user, so that it can be copied & pasted.

What do you think?

Reply:
Indeed this is what I initially challenged @Mcastres with on this but this feature will only allow setting the license.txt in development mode. In production/staging it just shows the license information.